### PR TITLE
[freerdp] initial integration

### PR DESCRIPTION
**Pull request by ligurio**
Issue: https://github.com/FreeRDP/FreeRDP/issues/5063
How To Build: https://github.com/awakecoding/FreeRDP-Manuals/blob/master/Developer/FreeRDP-Developer-Manual.markdown

### CODE REVIEW

File: projects/freerdp/project.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+homepage: "https://www.freerdp.com/"
+language: c
+primary_contact: ""
+builds_per_day: 4
+auto_ccs:
+  - "estetus@gmail.com"
+fuzzing_engines:
+  - afl
+  - honggfuzzer
```
Comment (jonathanmetzman) [inline projects/freerdp/project.yaml:9]: It's "honggfuzz" not "honggfuzzer"

File: projects/freerdp/project.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  - libfuzzer
+sanitizers:
+  - address
+  - memory
```
Comment (jonathanmetzman) [inline projects/freerdp/project.yaml:13]: "memory" sanitizer is generally hard to get working, unlike "undefined" behavior sanitizer, is there a reason why you don't want to try getting "undefined" working?

File: projects/freerdp/project.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+architectures:
+  - x86_64
+  - i386
```
Comment (jonathanmetzman) [inline projects/freerdp/project.yaml:16]: "i386" fuzzing is super optional. I'd try to get x86_64 fuzzing working first.

**Review by jonathanmetzman — COMMENTED** [review]
We approve of adding this project, just please fix the build errors first: https://github.com/google/oss-fuzz/pull/4748/checks?check_run_id=1474485173
I'll merge once you've fixed them.

**Comment by ligurio** [comment]
@jonathanmetzman Thanks for review! Although it was a raw patch. Unfortunately there is a problem with static build for tests. Still fighting with it.

File: projects/freerdp/project.yaml (hunk 1 above)
Comment (ligurio) [reply projects/freerdp/project.yaml]: Fixed.

File: projects/freerdp/project.yaml (hunk 2 above)
Comment (ligurio) [reply projects/freerdp/project.yaml]: replaced `memory` with `undefined`

File: projects/freerdp/project.yaml (hunk 3 above)
Comment (ligurio) [reply projects/freerdp/project.yaml]: removed `i386`

**Comment by jonathanmetzman** [comment]
There are some is

> @jonathanmetzman Thanks for review! Although it was a raw patch. Unfortunately there is a problem with static build for tests. Still fighting with it.

Ah sorry!

**Comment by ligurio** [comment]
@jonathanmetzman AFL exits with "no instrumentation in binary files", I have no idea what compiler option it is needed.
CFLAGS and CXXFLAGS successfully passed to source files as it is required by OSS Fuzz. Could you help with it?

**Comment by akallabeth** [comment]
@ligurio as discussed, please use team@freerdp.com (google account) as contact.

**Comment by oliverchang** [comment]
Sorry about that! I was cleaning out some old issues/PRs, didn't realise this one was still active.

**Comment by jonathanmetzman** [comment]
I don't really know what could cause this. Maybe reducing the number of threads used during linking would help.
The only other fix I can think of is rolling clang

**Comment by ligurio** [comment]
Waiting for merging https://github.com/FreeRDP/FreeRDP/pull/8989

**Comment by ligurio** [comment]
> Waiting for merging https://github.com/FreeRDP/FreeRDP/pull/8989

Merged.

**Comment by oliverchang** [comment]
CC @akallabeth from the linked PR in FreeRDP. Confirming that you are OK with this project's inclusion, and including @ligurio in any bug reports that are generated?

**Comment by akallabeth** [comment]
@oliverchang feel free to include this, all checks for issues are welcome by us :+1:

**Review by oliverchang — APPROVED** [review]

**Comment by akallabeth** [comment]
@oliverchang is it possible to give [freerdpproject@gmail.com](mailto:freerdpproject@gmail.com) access to this as well?

**Pull request merged**